src/main/scala/strawman/collection/IterableFactory.scala

```diff
@@ -0,0 +1,35 @@
+package strawman
+package collection
+
+import strawman.collection.mutable.Builder
+
+import scala.Int
+
+/** Base trait for instances that can construct a collection from an iterable */
+trait FromIterable[+C[X] <: Iterable[X]] {
+  def fromIterable[B](it: Iterable[B]): C[B]
+}
+
+/** Base trait for builder factories */
+trait CanBuild[-A, +Repr] {
```

Comment:
I find a little bit nicer to have implicit vals of type `CanBuild[…]` rather than `() => Builder[…]`. I find it also a little bit nicer to take parameters of type `CanBuild[…]` rather than `() => Builder[…]`. This is a matter of taste…